src/containerapp/azext_containerapp/_ssh_utils.py

```diff
@@ -0,0 +1,169 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=logging-fstring-interpolation
+# pylint: disable=possibly-used-before-assignment
+
+import sys
+import time
+import threading
+
+import websocket
+from azure.cli.command_modules.containerapp._ssh_utils import SSH_PROXY_INFO, SSH_DEFAULT_ENCODING, SSH_PROXY_ERROR, \
+    SSH_PROXY_FORWARD, SSH_CLUSTER_STDOUT, SSH_CLUSTER_STDERR, SSH_TERM_RESIZE_PREFIX, SSH_INPUT_PREFIX
+from azure.cli.command_modules.containerapp._utils import is_platform_windows
+from azure.cli.core.azclierror import CLIInternalError, ValidationError
+from azure.cli.core.commands.client_factory import get_subscription_id
+
+from knack.log import get_logger
+
+from ._clients import ContainerAppClient
+
+# pylint: disable=import-error,ungrouped-imports
+if is_platform_windows():
+    import msvcrt
+    from azure.cli.command_modules.container._vt_helper import (enable_vt_mode, _get_conout_mode,
+                                                                _set_conout_mode, _get_conin_mode, _set_conin_mode)
+
+logger = get_logger(__name__)
+
+
+class WebSocketConnection:
+    def __init__(self, cmd, resource_group_name, name, revision, replica, container):
+        token_response = ContainerAppClient.get_auth_token(cmd, resource_group_name, name)
+        self._token = token_response["properties"]["token"]
```

Comment:
Do we need some kind of token refresh logic to ensure the session won't be killed because of this token is expired?

Reply:
The token expire could prevent some security issues. For example - when I left my pc sometime, someone input something on the terminal, that's very terrible.
So I think no need to do some token refresh logic, this is helpful in some scenarios.